lib/tools.js

```diff
@@ -2204,6 +2204,48 @@ function getInstances(adapter, objects, withObjects, callback) {
     });
 }
 
+/**
+ * Checks if the given callback is a function and if so calls it with the given parameter immediately, else a resolved Promise is returned
+ *
+ * @param {function(...any=):void|undefined} callback - callback function to be executed
+ * @param {...any} [args] - as many arguments as needed, which will be returned by the callback function or by the Promise
+ * @returns {Promise} - if Promise is resolved with multiple arguments, an array is returned
+ */
+function maybeCallback(callback, args) {
+    args = Array.from(arguments).slice(1);
```

Comment:
I'll do an extensive review later, but you could just use rest arguments here:
```
function maybeCallback(callback, ...args) {
  // no slicing arguments
  // ...rest
}
```

Reply:
Thanks, this is exactly what I was looking for. ;-)